### ISSUE TITLE
Ability to call different content dynamically in elementor / Request a new feature

### ISSUE THREAD

**Issue by 79ho3ein** [elementor/elementor#16450]
- [ ] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [ ] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**

hello
There is a problem with many websites that we all (website designers) have definitely dealt with...

When we design a website for customers, we definitely put information such as address, email, location, contact number, business hours of the customer in different parts of the site. This information may sometimes need to be changed for any reason.

**But what is the problem?**
When we design the website and deliver it to the customer, the customer may change their contact number for any reason after a while. Either move your business to another location or even change your business hours. But she can not change this information,That is, it cannot enter the elementor editor, and change every single piece of information. because she does not have the knowledge to do so. It may also not have enough time to do so.

**What is the offer?**
I suggest you consider a section in the WordPress counter > elementor tab. Using this section, we call the required information dynamically. For example working hours.
For example, suppose the contact number is in the header, footer, contact us tab, about us, and even a few other tabs. Just change the contact number in the WordPress counter> elementor tab. Therefore, the contact number changes in all parts of the website.
With this feature, we no longer need to enter individual tabs to be able to change information. Once, all at once And this is wonderful.
It should be noted that this feature is currently available in the jet engine plugin. It would be great if you localized this in elementor.

thank you

**Comment by DavidGMiles** [comment]
That sounds like the standard WordPress options page functionality and then dropping the information in via shortcode - it's easily done

**Comment by 79ho3ein** [comment]
hi @DavidGMiles, No, this is not possible at the moment.
This feature is actually called **options page**.
Please watch this video : https://www.youtube.com/watch?v=MqcuHrT1a_c

**Comment by DavidGMiles** [comment]
Yup - standard WP feature - https://codex.wordpress.org/Creating_Options_Pages

JetEngine have just added a lot of functionality that you can do with ACF or MetaBox or hand code in WP

It's been in WordPress itself for 6 years at least

**Comment by DavidGMiles** [comment]
This is an example of setting up a settings page - or options page - in code

/**
 * Generated by the WordPress Option Page generator
 * at http://jeremyhixon.com/wp-tools/option-page/
 */

class SiteSettings {
	private $site_settings_options;

	public function __construct() {
		add_action( 'admin_menu', array( $this, 'site_settings_add_plugin_page' ) );
		add_action( 'admin_init', array( $this, 'site_settings_page_init' ) );
	}

	public function site_settings_add_plugin_page() {
		add_menu_page(
			'Site Settings', // page_title
			'Site Settings', // menu_title
			'manage_options', // capability
			'site-settings', // menu_slug
			array( $this, 'site_settings_create_admin_page' ), // function
			'dashicons-admin-generic', // icon_url
			2 // position
		);
	}

	public function site_settings_create_admin_page() {
		$this->site_settings_options = get_option( 'site_settings_option_name' ); ?>

		<div class="wrap">
			<h2>Site Settings</h2>
			<p></p>
			<?php settings_errors(); ?>

			<form method="post" action="options.php">
				<?php
					settings_fields( 'site_settings_option_group' );
					do_settings_sections( 'site-settings-admin' );
					submit_button();
				?>
			</form>
		</div>
	<?php }

	public function site_settings_page_init() {
		register_setting(
			'site_settings_option_group', // option_group
			'site_settings_option_name', // option_name
			array( $this, 'site_settings_sanitize' ) // sanitize_callback
		);

		add_settings_section(
			'site_settings_setting_section', // id
			'Settings', // title
			array( $this, 'site_settings_section_info' ), // callback
			'site-settings-admin' // page
		);

		add_settings_field(
			'telephone_number_0', // id
			'Telephone Number', // title
			array( $this, 'telephone_number_0_callback' ), // callback
			'site-settings-admin', // page
			'site_settings_setting_section' // section
		);

		add_settings_field(
			'address_1', // id
			'Address', // title
			array( $this, 'address_1_callback' ), // callback
			'site-settings-admin', // page
			'site_settings_setting_section' // section
		);

		add_settings_field(
			'something_else_2', // id
			'Something Else', // title
			array( $this, 'something_else_2_callback' ), // callback
			'site-settings-admin', // page
			'site_settings_setting_section' // section
		);
	}

	public function site_settings_sanitize($input) {
		$sanitary_values = array();
		if ( isset( $input['telephone_number_0'] ) ) {
			$sanitary_values['telephone_number_0'] = sanitize_text_field( $input['telephone_number_0'] );
		}

		if ( isset( $input['address_1'] ) ) {
			$sanitary_values['address_1'] = esc_textarea( $input['address_1'] );
		}

		if ( isset( $input['something_else_2'] ) ) {
			$sanitary_values['something_else_2'] = $input['something_else_2'];
		}

		return $sanitary_values;
	}

	public function site_settings_section_info() {
		
	}

	public function telephone_number_0_callback() {
		printf(
			'<input class="regular-text" type="text" name="site_settings_option_name[telephone_number_0]" id="telephone_number_0" value="%s">',
			isset( $this->site_settings_options['telephone_number_0'] ) ? esc_attr( $this->site_settings_options['telephone_number_0']) : ''
		);
	}

	public function address_1_callback() {
		printf(
			'<textarea class="large-text" rows="5" name="site_settings_option_name[address_1]" id="address_1">%s</textarea>',
			isset( $this->site_settings_options['address_1'] ) ? esc_attr( $this->site_settings_options['address_1']) : ''
		);
	}

	public function something_else_2_callback() {
		printf(
			'<input type="checkbox" name="site_settings_option_name[something_else_2]" id="something_else_2" value="something_else_2" %s> <label for="something_else_2">Something</label>',
			( isset( $this->site_settings_options['something_else_2'] ) && $this->site_settings_options['something_else_2'] === 'something_else_2' ) ? 'checked' : ''
		);
	}

}
if ( is_admin() )
	$site_settings = new SiteSettings();

/* 
 * Retrieve this value with:
 * $site_settings_options = get_option( 'site_settings_option_name' ); // Array of All Options
 * $telephone_number_0 = $site_settings_options['telephone_number_0']; // Telephone Number
 * $address_1 = $site_settings_options['address_1']; // Address
 * $something_else_2 = $site_settings_options['something_else_2']; // Something Else
 */

**Comment by 79ho3ein** [comment]
thank you
